core/domain/event_services.py

```diff
@@ -75,35 +76,21 @@ def _notify_continuous_computation_listeners_async(cls, *args, **kwargs):
 
     @classmethod
     def _handle_event(cls, exploration_id, exploration_version, state_name,
-            rule_spec_string, session_id, time_spent_in_secs, params, answer):
-        """Records an event when an answer triggers a rule."""
+            answer_group_index, rule_spec_index, classification_categorization,
+            session_id, time_spent_in_secs, params, normalized_answer):
+        """Records an event when an answer triggers a rule. The answer recorded
+        here is a Python-representation of the actual answer submitted by the
+        user.
+        """
         # TODO(sll): Escape these args?
-        # TODO(msl): remove old answer recording models.
-
-        # In the old framework, answers are converted to unicode even when they
-        # are not strings. This will be removed when migrating to the new
-        # framework.
-        stats_models.process_submitted_answer(
-            exploration_id, exploration_version, state_name,
-            rule_spec_string, unicode(answer))
+        exploration = exp_services.get_exploration_by_id(
```

Comment:
You could perhaps calculate the interaction_id in the controller, and pass it from there instead.